src/som/interpreter/actors/Actor.java

```diff
@@ -182,11 +187,34 @@ protected void appendToMailbox(final EventualMessage msg) {
     mailboxExtension.append(msg);
   }
 
-  protected static void handleBreakPoints(final EventualMessage msg, final WebDebugger dbg) {
-    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED && msg.isBreakpoint()) {
-      dbg.prepareSteppingUntilNextRootNode();
+  protected static void handleBreakpointsAndStepping(final EventualMessage msg, final WebDebugger dbg) {
+    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+      // check step-into and breakpoint before the message is processed
+      if (msg.isStepInto()) {
+        dbg.prepareStepIntoMessage();
+      } else if (msg.isBreakpoint()) {
```

Comment:
`isBreakpoint()` sounds now very unspecific, I think it would be good to rename to clarify what it actually checks.
Also, it is unclear to me that `isBreakpoint()` and `isStepInto()` two are truly mutually exclusive.